homeassistant/components/rfxtrx/config_flow.py

```diff
@@ -1,12 +1,296 @@
+            if CONF_REMOVE_DEVICE in user_input:
+                device_data = self._get_device_data(user_input[CONF_REMOVE_DEVICE])
+
+                event_code = device_data[CONF_EVENT_CODE]
+                device_id = device_data[CONF_DEVICE_ID]
+                self.hass.helpers.dispatcher.async_dispatcher_send(
+                    f"{DOMAIN}_{CONF_REMOVE_DEVICE}_{device_id}"
+                )
+                self._device_registry.async_remove_device(
+                    user_input[CONF_REMOVE_DEVICE]
+                )
+                devices = {event_code: None}
+                self.update_config_data(
+                    global_options=self._global_options, devices=devices
+                )
+
+                return self.async_create_entry(title="", data={})
+            if CONF_EVENT_CODE in user_input:
+                self._selected_device_event_code = user_input[CONF_EVENT_CODE]
+                self._selected_device = {}
+                selected_device_object = get_rfx_object(
+                    self._selected_device_event_code
+                )
+                if selected_device_object is None:
+                    errors[CONF_EVENT_CODE] = "invalid_event_code"
+                else:
+                    self._selected_device_object = selected_device_object
+                    return await self.async_step_set_device_options()
+
+            if not errors:
+                self.update_config_data(global_options=self._global_options)
+
+                return self.async_create_entry(title="", data={})
+
+        device_registry = await async_get_registry(self.hass)
+        device_entries = async_entries_for_config_entry(
+            device_registry, self._config_entry.entry_id
+        )
+        self._device_registry = device_registry
+        self._device_entries = device_entries
+
+        devices = {
+            entry.id: entry.name_by_user if entry.name_by_user else entry.name
+            for entry in device_entries
+        }
+
+        options = {
+            vol.Optional(CONF_DEBUG, default=self._config_entry.data[CONF_DEBUG]): bool,
+            vol.Optional(
+                CONF_AUTOMATIC_ADD, default=self._config_entry.data[CONF_AUTOMATIC_ADD],
+            ): bool,
+            vol.Optional(CONF_EVENT_CODE): str,
+            vol.Optional(CONF_DEVICE): vol.In(devices),
+            vol.Optional(CONF_REMOVE_DEVICE): vol.In(devices),
+        }
+
+        return self.async_show_form(
+            step_id="prompt_options", data_schema=vol.Schema(options), errors=errors
+        )
+
+    async def async_step_set_device_options(self, user_input=None):
+        """Manage device options."""
+        errors = {}
+
+        if user_input is not None:
+            device_id = get_device_id(
+                self._selected_device_object.device,
+                data_bits=user_input.get(CONF_DATA_BITS),
+            )
+
+            def none_or_int(value, base):
```

Comment:
We can move this function to the module level.

Reply:
Where do you want to move it? (What would be the module level)

Reply:
Either after the constants or at the end of this module.